crates/red_knot_python_semantic/src/symbol.rs

```diff
@@ -541,8 +541,8 @@ fn symbol_from_bindings_impl<'db>(
                 return None;
             }
 
-            let mut constraint_tys = narrowing_constraints
-                .filter_map(|constraint| narrowing_constraint(db, constraint, binding))
+            let mut constraint_tys = narrowing_constraint
+                .filter_map(|constraint| infer_narrowing_constraint(db, constraint, binding))
```

Comment:
Coming back to the terminology problem: here we iterate over `narrowing_constraint`, and the iteration yields something we also call a `constraint`, which we then pass to a function named `infer_narrowing_constraint`.